console/src/tasks.rs

```diff
@@ -127,12 +164,23 @@ impl State {
             let fields = task
                 .fields
                 .iter()
-                .fold(String::new(), |mut res, f| {
-                    write!(&mut res, "{} ", f).unwrap();
-                    res
+                .filter_map(|f| {
+                    let field_name = f.name.as_ref().expect("no name");
+                    let name = match field_name {
+                        proto::field::Name::StrName(n) => Some(n.clone()),
+                        proto::field::Name::NameIdx(idx) => {
+                            let meta_id = f.metadata_id.as_ref().expect("no id");
+                            metas
+                                .get(&meta_id.id)
+                                .and_then(|meta| meta.field_names.get(*idx as usize))
+                                .cloned()
+                        }
+                    };
+                    let value = f.value.as_ref().expect("no value").clone().into();
+                    name.map(|name| Field { name, value })
```

Comment:
we never reuse the protobuf `Task`'s fields, so we shouldn't have to clone them here. if we moved the fields by value, we could write something like this instead:
```rust
            let fields = task
                .fields
                .drain(..)
                .filter_map(|f| {
                    let field_name = f.name.expect("no name");
                    let name = match field_name {
                        proto::field::Name::StrName(n) => Some(n),
                        proto::field::Name::NameIdx(idx) => {
                            let meta_id = f.metadata_id.as_ref().expect("no id");
                            metas
                                .get(&meta_id.id)
                                .and_then(|meta| meta.field_names.get(*idx as usize))
                                .cloned()
                        }
                    };
                    let value = f.value.expect("no value").into();
                    name.map(|name| Field { name, value })
                })
                .collect();
```